packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -120,54 +133,139 @@ export type EditableTreeOrPrimitive = EditableTree | PrimitiveValue;
+    /**
+     * The `FieldSchema` of this field.
+     */
+    readonly fieldSchema: FieldSchema;
+
+    /**
+     * The `FieldKey` of this field.
+     */
+    readonly fieldKey: FieldKey;
+
+    /**
+     * Gets the type of the node by its index or the "field type" if index is not provided and this is the primary field.
+     * If the node is well-formed, it must follow its schema.
+     * @param index - if index is provided, returns the type of the node. Otherwise, if this field is the primary field,
+     * returns the type of the parent node this primary field belongs to.
+     * @param nameOnly - if true, returns only the type identifier.
+     */
+    getType(index?: number, nameOnly?: boolean): NamedTreeSchema | TreeSchemaIdentifier | undefined;
+
+    /**
+     * Gets a node of this field by its index without unwrapping.
+     * Note that the node must exists at the given index.
+     */
+    getWithoutUnwrapping(index: number): EditableTree;
+
+    /**
+     * Stores the target for the proxy which implements reading and writing for this sequence field.
+     * The details of this object are implementation details,
+     * but the presence of this symbol can be used to separate EditableTrees from other types.
+     */
+    readonly [proxyTargetSymbol]: object;
+
+    /**
+     * Gets an iterator iterating over the nodes (unwrapped) of this field.
+     * See {@link UnwrappedEditableTree} for what does "unwrapped" mean.
+     * It reads all nodes at once before the iteration starts to get a "snapshot" of this field.
+     * It might be inefficient regarding resources, but avoids situations
+     * when the field is getting changed while iterating.
+     */
+    [Symbol.iterator](): IterableIterator<UnwrappedEditableTree>;
+}
 
 /**
  * Unwrapped field.
  * Non-sequence multiplicities are unwrapped to the child tree or `undefined` if there is none.
- * Sequence multiplicities are handled with arrays.
+ * Sequence multiplicities are handled with {@link EditableField}.
  * See {@link UnwrappedEditableTree} for how the children themselves are unwrapped.
  */
-export type UnwrappedEditableField =
-    | UnwrappedEditableTree
-    | undefined
-    | readonly UnwrappedEditableTree[];
+export type UnwrappedEditableField = UnwrappedEditableTree | undefined | EditableField;
 
-export class ProxyTarget {
+/**
+ * This is a base class for `ProxyTarget` and `SequenceProxyTarget`, which uniformly handles cursors and anchors.
+ */
+export abstract class BaseProxyTarget {
     private readonly lazyCursor: ITreeSubscriptionCursor;
-    private anchor?: Anchor;
-
-    constructor(public readonly context: ProxyContext, cursor: ITreeSubscriptionCursor) {
-        this.lazyCursor = cursor.fork();
-        context.withCursors.add(this);
+    private anchor?: Anchor | FieldAnchor;
+
+    constructor(
+        public readonly context: ProxyContext,
+        cursor: ITreeSubscriptionCursor,
+        public readonly fieldKey: FieldKey,
```

Comment:
this should not need to take in a fieldKey: the cursor should be enough.

We should not need to special case rootFieldKeySymbol in here either. We can just pass in cursors for the root fields, and it should just work.

Also note that forest can (transiently) have roots other than rootFieldKeySymbol: I suspect the code below might not handle them correctly.